Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -614,6 +616,44 @@ void NetworkStorageManager::persisted(const WebCore::ClientOrigin& origin, Compl
     completionHandler(persistedInternal(origin));
 }
 
+void NetworkStorageManager::fetchRegistrableDomainsForPersist()
+{
+    ASSERT(RunLoop::isMain());
+
+    if (!m_process)
+        return didFetchRegistrableDomainsForPersist({ });
+
+    m_process->registrableDomainsExemptFromWebsiteDataDeletion(m_sessionID, [weakThis = ThreadSafeWeakPtr { *this }](auto&& domains) mutable {
+        if (auto strongThis = weakThis.get())
+            strongThis->didFetchRegistrableDomainsForPersist(WTFMove(domains));
+    });
+}
+
+void NetworkStorageManager::didFetchRegistrableDomainsForPersist(HashSet<WebCore::RegistrableDomain>&& domains)
+{
+    ASSERT(RunLoop::isMain());
+
+    if (m_closed)
+        return;
+
+    m_queue->dispatch([this, weakThis = ThreadSafeWeakPtr { *this }, domains = crossThreadCopy(WTFMove(domains))]() mutable {
+        assertIsCurrent(workQueue());
+
+        auto strongThis = weakThis.get();
+        if (!strongThis)
+            return;
+
+        for (auto&& [origin, completionHandler] : std::exchange(m_persistCompletionHandlers, { })) {
+            if (!domains.contains(origin.clientRegistrableDomain()))
+                completionHandler(false);
```

Comment:
In theory, we should delete the file if it exists.

Reply:
Okay, will update.